✨(edxapp) create required directories in volumes and release Arnold 4.6.0

sampaccoud opened the pull request:
## Purpose

Some directories must pre-exist inside edxapp's volumes for the code to work. 

## Proposal

We propose to add a job that takes care of this on the basis of a variable listing all such required directories. The list of directories can thus be overriden for each project.

sampaccoud committed:
✨(edxapp) create required directories in volumes

Some directories must pre-exist inside edxapp's volumes for the code
to work. We propose to add a job that takes care of this on the basis
of a variable listing all such required directories. The list of
directories can thus be overriden for each project.

jmaupetit commented on apps/edxapp/templates/services/cms/job_01_create_directories.yml.j2:
```diff
@@ -0,0 +1,47 @@
+          command:
+            - "bash"
+            - "-c"
+            - {% for path in required_directories %}(mkdir {{ path }} || true) && {% endfor %}echo "✅"
```
You should use `-p` in case of nested, deep paths and it always returns a success exit code.
```suggestion
            - {% for path in required_directories %}mkdir -p {{ path }} && {% endfor %}echo "✅"
```

sampaccoud replied:
:medal_sports:

jmaupetit commented on apps/edxapp/templates/services/cms/job_01_create_directories.yml.j2:
```diff
@@ -0,0 +1,47 @@
+            claimName: edxapp-pvc-media
+        - name: edxapp-v-data
+          persistentVolumeClaim:
+            claimName: edxapp-pvc-data
```
I think we should mount all available volumes here?

sampaccoud replied:
you're right I mounted what I needed but since required directories= is configurable we should mount all.

jmaupetit commented on apps/edxapp/vars/all/main.yml:
```diff
@@ -72,6 +72,8 @@ edxapp_export_volume_size: 100Mi
 
 # -- volume names
 edxapp_export_volume_path: /edx/var/edxapp/export
+required_directories:
```
Should be prefixed by the app name?
```suggestion
edxapp_required_directories:
```

sampaccoud replied:
oh yes! So obvious I couldn't see it ^^

sampaccoud committed:
🔖(minor) bump release to 4.6.0

Added

- Job in edxapp to create required directories in volumes
- Whitelist ips authorized to access /admin on edxapp,
  edxec, marsha and richie
- Whitelist ips authorized to access learninglocker's UI

Fixed

- In edxapp, add an lms queue to the list of cms Celery queues
- Add edxec to eugene/development apps list
